src/dotnet/Chat.UI.Blazor/Components/ChatView/ChatView.razor.cs

```diff
@@ -233,4 +229,17 @@ private Task OnNavigateToChatEntry(NavigateToChatEntry navigation, CancellationT
         NavigateToEntry(navigation.ChatEntryId);
         return Task.CompletedTask;
     }
+
+    private void OnLocationChanged(object? sender, LocationChangedEventArgs e)
+        => TryNavigateToEntryIfExists();
+
+    private void TryNavigateToEntryIfExists()
+    {
+        var currentUri = new Uri(Nav.Uri);
+        var entryIdString = currentUri.Fragment?.TrimStart('#');
+        if (long.TryParse(entryIdString, out var entryId) && entryId > 0) {
+            NavigateToEntry(entryId);
+            ChatUI.HighlightedChatEntryId.Value = entryId;
```

Comment:
I see here a room for improvement. But apparently it better to do in a separate task.
ChatUI.HighlightedChatEntryId.Value is reseted to 0L automatically in 2 seconds. If chat entry we are navigating to is not loaded yet and it takes significant amout of time to bring it to viewport and displey, it may happen that highlighing will last shorter than 2 seconds or not visible at all.